roles/kubernetes/preinstall/tasks/0020-verify-settings.yml

```diff
@@ -334,3 +334,18 @@
   when:
     - containerd_config is defined
     - not ignore_assert_errors
+
+- name: Get current state of time synchronization
+  shell: "echo $(($(date +%s%N)/1000000))"
+  args:
+    executable: /bin/bash
+  changed_when: false
+  register: cur_time_ms_string
+
+- name: Stop if time is not synchronized
+  assert:
+    that: ((hostvars[item]['cur_time_ms_string'].stdout|int - cur_time_ms_string.stdout|int) | abs) < 60000
+    msg: "Do not allow more than 60s diff between servers, {{ hostvars[item]['inventory_hostname'] }} time is different."
```

Comment:
If we're adding this, wouldn't we want to make the allowed drift a configurable value? 60s seems quite high to me for example (although it might conversely seems too low in some environments perhaps)

Reply:
HI  @Tristan971 , the PR is reopened at https://github.com/kubernetes-sigs/kubespray/pull/8824. Would you please review it?